File: apps/desktop/src/api/evidence-review.ts
import { apiGet, apiPatch, apiPost } from "./client"
import type {
  AgentLiteratureSynthesisResponse,
  EvidenceReviewSnapshot,
  EvidenceReviewStatus,
  LiteratureSynthesisPlan,
  ReviewedEvidenceItem,
} from "../features/research/evidence-review-types"

export function getEvidenceReview(
  workspaceId: string,
  query = "",
  limit = 100,
): Promise<EvidenceReviewSnapshot> {
  const params = new URLSearchParams({ q: query, limit: String(limit) })
  return apiGet<EvidenceReviewSnapshot>(
    `/api/research/workspaces/${encodeURIComponent(workspaceId)}/evidence-review?${params.toString()}`,
  )
}

export function updateEvidenceReview(
  workspaceId: string,
  entryId: string,
  status: EvidenceReviewStatus,
  note = "",
): Promise<ReviewedEvidenceItem> {
  return apiPatch<ReviewedEvidenceItem, { status: EvidenceReviewStatus; note: string }>(
    `/api/research/workspaces/${encodeURIComponent(workspaceId)}/evidence-review/${encodeURIComponent(entryId)}`,
    { status, note },
  )
}

export function synthesizeLiterature(
  workspaceId: string,
  query = "",
): Promise<LiteratureSynthesisPlan> {
  return apiPost<LiteratureSynthesisPlan, { query: string }>(
    `/api/research/workspaces/${encodeURIComponent(workspaceId)}/literature-synthesis`,
    { query },
  )
}

export function synthesizeLiteratureWithAgent(
  workspaceId: string,
  query = "",
): Promise<AgentLiteratureSynthesisResponse> {
  return apiPost<AgentLiteratureSynthesisResponse, { query: string }>(
    `/api/research/workspaces/${encodeURIComponent(workspaceId)}/literature-synthesis/agent`,
    { query },
  )
}

File: backend/api/evidence_review.py
from __future__ import annotations

from typing import Annotated

from fastapi import APIRouter, Depends, HTTPException, Query, status

from backend.api.evidence_review_dependencies import get_evidence_review_service
from backend.models.evidence_review import (
    EvidenceReviewSnapshot,
    EvidenceReviewUpdateRequest,
    LiteratureSynthesisPlan,
    LiteratureSynthesisRequest,
    ReviewedEvidenceLedgerItem,
)
from backend.services.evidence_review_service import EvidenceReviewService

router = APIRouter(prefix="/api/research", tags=["evidence-review"])
EvidenceReviewDependency = Annotated[EvidenceReviewService, Depends(get_evidence_review_service)]


def _raise_review_error(exc: Exception) -> None:
    message = str(exc) or "Invalid Evidence Review request."
    code = status.HTTP_404_NOT_FOUND if "not found" in message.casefold() else status.HTTP_422_UNPROCESSABLE_ENTITY
    raise HTTPException(status_code=code, detail=message) from exc


@router.get("/workspaces/{workspace_id}/evidence-review", response_model=EvidenceReviewSnapshot)
def get_evidence_review(
    workspace_id: str,
    service: EvidenceReviewDependency,
    q: str = Query(default="", max_length=4000),
    limit: int = Query(default=100, ge=1, le=500),
) -> EvidenceReviewSnapshot:
    try:
        return service.snapshot(workspace_id=workspace_id, query=q, limit=limit)
    except Exception as exc:  # noqa: BLE001
        _raise_review_error(exc)
        raise AssertionError("unreachable")


@router.patch(
    "/workspaces/{workspace_id}/evidence-review/{entry_id}",
    response_model=ReviewedEvidenceLedgerItem,
)
def review_evidence_entry(
    workspace_id: str,
    entry_id: str,
    payload: EvidenceReviewUpdateRequest,
    service: EvidenceReviewDependency,
) -> ReviewedEvidenceLedgerItem:
    try:
        return service.review(
            workspace_id=workspace_id,
            entry_id=entry_id,
            status=payload.status,
            note=payload.note,
        )
    except Exception as exc:  # noqa: BLE001
        _raise_review_error(exc)
        raise AssertionError("unreachable")


@router.post(
    "/workspaces/{workspace_id}/literature-synthesis",
    response_model=LiteratureSynthesisPlan,
)
def synthesize_literature(
    workspace_id: str,
    payload: LiteratureSynthesisRequest,
    service: EvidenceReviewDependency,
) -> LiteratureSynthesisPlan:
    try:
        return service.synthesize(workspace_id=workspace_id, query=payload.query)
    except Exception as exc:  # noqa: BLE001
        _raise_review_error(exc)
        raise AssertionError("unreachable")


__all__ = ["router"]
